core/src/consensus.rs

```diff
@@ -139,6 +141,7 @@ pub(crate) struct ComputedBankState {
 #[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
 pub enum TowerVersions {
     V1_17_14(Tower1_7_14),
+    V1_14_11(Tower1_14_11),
```

Comment:
are the tower changes strictly required to reserve the new space in `VoteState`?  i'm not seeing any dependencies

we should defer them to the changes that implement the actual timely vote credits feature if so

Reply:
The issue is that if the tower modifications here are not made, then the serialized format of the tower changes when the VoteState struct is changed, since the tower stores a VoteState instance.  This then means that upgrading to the version of validator software that includes this change will start writing a tower file that on downgrade will be unreadable by the downgraded-to software.

Therefore I have altered the tower logic to continue to serialize the tower using the older serialized format.  I don't see any way around this if the goal of ensuring that downgrade is not broken is to be achieved.

Reply:
I guess an alternative would be to write the new tower file into a new location, similar to what was done with the tower-1.9 file.  Both versions of the tower (tower-1.9 and tower-1.14) would be written at the same time, until the old file is deprecated later.

However, this would still be changes to the tower, so it doesn't avoid code modifications.  In addition, it's a more complex change that is unnecessary because there is no new data needed in the tower file.  If the tower actually had new data then this would be the necessary approach.

So I was inaccurate when I said "I don't see any way around this", I should have said "I don't see a better way around this".

Reply:
Yeah, gonna be especially problematic when the 1.17 branch actually exists.